### PR TITLE
`fre pp configure-yaml` entry-point for click fix, test

### PR DESCRIPTION
**Pull request by ilaflott**
## Describe your changes

- click entry-point adjustments, particularly for `fre pp configure-yaml`
- some `chdir`s that werent needed are removed
- some `chdir`s that ARE needed are un-done by the end of the routine
- a `fre make` tests now re-sets the `os.environ['HOME']` back to what it was before adjustment
- adds a unit test that will catch a mal-functioning click entry-point for `fre pp configure-yaml`
- adds, FINALLY, a functional unit test for `fre pp checkout`

## Checklist before requesting a review

- [x] I ran my code
- [x] I tried to make my code readable
- [x] I tried to comment my code
- [x] I wrote a new test, if applicable
- [x] I wrote new instructions/documentation, if applicable
- [x] I ran pytest and inspected it's output
- [x] I ran pylint and attempted to implement some of it's feedback

### CODE REVIEW

**Comment by ilaflott** [comment]
thanks @rem1776 for pointing out the makefile test behavior, @kiihne-noaa for getting me (accidentally) to peek at this again for something totally different, and @cwhitlock-NOAA for your initial frustrations/attempts that i could not make sense of without @rem1776 's tip

File: fre/make/tests/test_create_makefile.py [hunk 1]
```diff
@@ -16,57 +16,60 @@
...
-    assert Path(f"{test_dir}/{NM_EXAMPLE}/platforms.yaml").exists()
+    assert Path(f"{TEST_DIR}/{NM_EXAMPLE}/platforms.yaml").exists()
 
 def test_bm_makefile_creation():
     """
     Check the makefile is created when a bare-metal platform is used
     """
     # Set output directory as home for fre make output
-    os.environ["HOME"]=str(Path(out))
+    def_home = str(os.environ["HOME"])
```
Comment (singhd789) [inline fre/make/tests/test_create_makefile.py:54]: Thank you for this! I was going to fix it today!

File: fre/pp/frepp.py [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 
 import click
-from .checkoutScript import checkoutTemplate
-from .configure_script_yaml import yamlInfo
+from .checkoutScript import _checkoutTemplate
+from .configure_script_yaml import _yamlInfo
 from .configure_script_xml import convert
-from .validate import validate_subtool
+from .validate import _validate_subtool
 from .install import install_subtool
```
Comment (singhd789) [inline fre/pp/frepp.py:8]: Do the click entry points for `install_subtool`, `pp_run_subtool`, and `status_subtool` need to be fixed too?

**Review by singhd789 — APPROVED** [review]

File: fre/pp/frepp.py (hunk 2 above)
Comment (ilaflott) [reply fre/pp/frepp.py:8]: yep, and they should follow the same clean-convention you followed for `fre make` as well! will address

File: fre/pp/checkoutScript.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def _checkoutTemplate(experiment, platform, target, branch='main'):
     os.makedirs(directory, exist_ok=True)
 
     # Change the current working directory
-    os.chdir(directory)
+    #os.chdir(directory)
```
Comment (ceblanton) [inline fre/pp/checkoutScript.py:28]: Nice.. we can remove it later

**Review by ceblanton — APPROVED** [review]
Thanks! Clear improvements all around.

**Pull request merged**